mungegithub/mungers/inactive-review-handler.go

```diff
@@ -0,0 +1,170 @@
+
+// AddFlags will add any request flags to the cobra `cmd`
+func (i *InactiveReviewHandler) AddFlags(cmd *cobra.Command, config *github.Config) {}
+
+// Determine whether the PR is active based on number of seconds
+// A PR is inactive if it is not being modified for more than one week
+// The munger will only run at most 5 times
+func (i *InactiveReviewHandler) isPRActive(prCreatedAt *time.Time, comments []*githubapi.IssueComment, reviewComments []*githubapi.PullRequestComment) bool {
+	pinger := matchers.NewPinger("INACTIVE-REVIEWER").SetTimePeriod(7 * 24 * time.Hour).SetMaxCount(5)
+	lastDate := matchers.Items{}.
+		AddComments(comments...).
+		AddReviewComments(reviewComments...).
+		Filter(matchers.HumanActor()).
+		LastDate(prCreatedAt)
+
+	return !pinger.IsMaxReached(comments, lastDate)
+}
+
+// Suggest a new reviewer who is NOT any of the existing reviewers
+// (1) get all current assignees for the PR
+// (2) get potential owners of the PR using Blunderbuss algorithm (calling getPotentialOwners() function)
+// (3) filter out current assignees from the potential owners
+// (4) if there is no any new reviewer available, the bot will encourage the PR author to ping all existing assignees
+// (5) otherwise, select a new reviewer using Blunderbuss algorithm (calling selectMultipleOwners() function with number of assignees parameter of one)
+// Note: the munger will suggest a new reviewer when the PR currently does not have any reviewer
+//		 the munger will only run at most 5 times
+func (i *InactiveReviewHandler) suggestNewReviewer(issue *githubapi.Issue, potentialOwners weightMap, weightSum int64) string {
+	var newReviewer string
+
+	if len(potentialOwners) > 0 && issue.Assignees != nil && len(issue.Assignees) > 0 {
+		for _, oldReviewer := range issue.Assignees {
+			login := *oldReviewer.Login
+
+			for potentialOwner := range potentialOwners {
+				if login == potentialOwner {
+					weightSum -= potentialOwners[login]
+					delete(potentialOwners, login)
+					break
+				}
+			}
+		}
+	}
+
+	if len(potentialOwners) == 0 {
+		return newReviewer
+	}
+
+	newReviewer = selectMultipleOwners(potentialOwners, weightSum, 1)[0]
+
+	return newReviewer
+}
+
+// Munge is the workhorse encouraging PR author to assign a new reviewer
+// after getting no response from current reviewer for one week
+// The algorithm:
+// (1) find last modification time of the reviewer
+// (2) if the date is one week or longer before today's date, create a comment
+//     encouraging the author to assign a new reviewer and unassign the old reviewer
+// (3) suggest the new reviewer using Blunderbuss algorithm, making sure the old reviewer is not suggested
+func (i *InactiveReviewHandler) Munge(obj *github.MungeObject) {
+	issue := obj.Issue
+
+	//do not suggest new reviewer if it is not a PR or the PR has no author information
+	if !obj.IsPR() || issue.User == nil || issue.User.Login == nil {
+		return
+	}
+
+	pr, ok := obj.GetPR()
```

Comment:
good idea! a closer look at the codebase shows that `pr.CreatedAt` and `obj.Issue.CreatedAt` are the same. thanks! :)